comment/tests/test_template_tags.py

```diff
@@ -144,6 +144,7 @@ def test_get_username_for_comment(self):
         anonymous_comment = self.create_anonymous_comment()
 
         self.assertEqual(get_username_for_comment(comment), comment.user.username)
+        patch.object(settings, 'COMMENT_USE_EMAIL_FIRST_PART_AS_USERNAME', False).start()
         self.assertEqual(get_username_for_comment(anonymous_comment), settings.COMMENT_ANONYMOUS_USERNAME)
 
         patch.object(settings, 'COMMENT_USE_EMAIL_FIRST_PART_AS_USERNAME', True).start()
```

Comment:
if different parts of the test use different values for the same attribute maybe when we can split them to separate functions or use a context-manager when patching to make the code more readable.

Reply:
You are right. We have to use patch decorator for changing settings value in all the test and avoid change settings values inside the test, that will be pain in the as* when a test fails due to a settings value.

Reply:
This did happen to me when I was working on the `pytest` branch, I will open an issue for this separately.